### PR TITLE
Alert reducer

### PR DESCRIPTION
**Pull request by WenxuC**
added alert reducer for when the two password does not match

### CODE REVIEW

**Review by channnx — APPROVED** [review]

File: client/src/actions/alert.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { v4 as uuidv4 } from 'uuid';
```
Comment (WenxuC) [inline client/src/actions/alert.js:1]: this package allows us to generate a random unique id for the alert

File: client/src/actions/types.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const SET_ALERT = 'SET_ALERT';
+export const REMOVE_ALERT = 'REMOVE_ALERT';
```
Comment (WenxuC) [inline client/src/actions/types.js:2]: this file stores all the CONST reducer types

File: client/src/components/Alert.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+			</div>
+		))}
+	</div>
+);
+
+Alert.propTypes = {
+	alerts: PropTypes.array.isRequired,
+};
+
+const mapStatetoProps = state => ({
```
Comment (WenxuC) [inline client/src/components/Alert.js:19]: gets the state of the reducer and map it to prop

File: client/src/components/Signup.js [hunk 4]
```diff
@@ -1,36 +1,43 @@
...
 
-function Signup() {
-	const handleSubmit = async event => {
-		event.preventDefault();
+const Signup = ({ setAlert }) => {
+	const [formData, setFormData] = useState({
+		userName: '',
+		email: '',
+		password: '',
+		password_confirm: '',
```
Comment (WenxuC) [inline client/src/components/Signup.js:26]: added another text field for password so we can make sure the user matches their password when signing up

File: client/src/components/Signup.js [hunk 5]
```diff
@@ -1,36 +1,43 @@
...
-		const res = await axios.post('http://localhost:8000/app/signup', newUser);
-		if (res.data.isRegistered) console.log('This email is no good');
-	};
+	const onChange = event =>
+		setFormData({ ...formData, [event.target.name]: event.target.value });
 
+	const onSubmit = async event => {
+		event.preventDefault();
+		if (password !== password_confirm) {
+			setAlert('Password does not match');
```
Comment (WenxuC) [inline client/src/components/Signup.js:37]: When the password doesn't match it throughs an alert, via the alert reducer

File: client/src/components/Signup.js [hunk 6]
```diff
@@ -1,36 +1,43 @@
...
+		}
+		console.log('Sign up complete', 'danger');
```
Comment (WenxuC) [inline client/src/components/Signup.js:39]: I will remove this in the future

**Pull request merged**